src/Microsoft.Framework.PackageManager/SourceControl/GitSourceControlProvider.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            if (!snapshotInformation.ContainsKey(RepoUrlKey) || string.IsNullOrEmpty(snapshotInformation[RepoUrlKey]))
+            {
+                throw new ArgumentNullException("The repository URL must be specified.");
+            }
+
+            if (!snapshotInformation.ContainsKey(CommitHashKey))
+            {
+                snapshotInformation[CommitHashKey] = GetHEADCommitId(folderName);
+            }
+
+            if (!snapshotInformation.ContainsKey(ProjectPathKey))
+            {
+                var repoRoot = GetRepoRoot(folderName);
+                var pathRelativeToRepoRoot = Path.GetDirectoryName(folderName)
+                    .Substring(repoRoot.Length)
+                    .Replace('\\', '/')
+                    .TrimStart('/');
+                snapshotInformation[ProjectPathKey] = pathRelativeToRepoRoot;
+            }
+        }
+
+        private string GetHEADCommitId(string folderName)
+        {
+            string stdOut;
+            string stdErr;
+
+            if (ProcessUtilities.RunApp(RepoApp, "rev-parse HEAD", folderName, out stdOut, out stdErr) &&
+                string.IsNullOrEmpty(stdErr))
+            {
+                return stdOut;
+            }
+
+            throw new InvalidOperationException(stdErr);
+        }
+
+        private string GetRepoRoot(string folderName)
+        {
+            string stdOut;
+            string stdErr;
+
+            if (ProcessUtilities.RunApp(RepoApp, "rev-parse --show-toplevel", folderName, out stdOut, out stdErr) &&
+                string.IsNullOrEmpty(stdErr))
+            {
+                return stdOut;
+            }
+
+            throw new InvalidOperationException(stdErr);
+        }
+
+        public override string CreateShortFolderName(IDictionary<string, string> snapshotInfo)
+        {
+            var repoUrl = snapshotInfo[RepoUrlKey];
+            var commitHash = snapshotInfo[CommitHashKey];
+
+            string repoName = Path.GetFileNameWithoutExtension(repoUrl);
+            string shortHash = commitHash.Substring(0, 8);
+
+            return repoName + shortHash;
+        }
+
+        public override bool GetSources(string destinationFolder, IDictionary<string, string> snapshotInfo)
+        {
+            var repoUrl = snapshotInfo[RepoUrlKey];
+            var commitHash = snapshotInfo[CommitHashKey];
+
+            string stdOut;
+            string stdErr;
+
+            _buildReports.WriteInformation($"Cloning from: {repoUrl}");
+
+            // First clone
+            if (!ProcessUtilities.RunApp(
+                RepoApp, 
+                $"clone {repoUrl} {destinationFolder}",
+                workingDirectory: null,
+                stdOut: out stdOut,
+                stdErr: out stdErr))
+            {
+                _buildReports.WriteError(stdErr);
+                return false;
```

Comment:
So if I do `dnu source MyPackage` for one project and then do `dnu source MyPackage` for another project that uses exactly same version of `MyPackage`, it will fail with
```
fatal: destination path '%userprofile%/.dnx/source/MyPackage-{CommitHash}' already exists and is not an empty directory.
```
Seems this is inconvenient when you switch between multiple projects.